systers/FirstAide-iOS
# add name labels to the Circle of Trust view controller

#### Issue

#43 - A user should be able to see labels below each character icon in the Circle of Trust to display each person's name. This will help the user remember who to contact in the case of an emergency.

#### Design mocks 

[Design Mocks](https://docs.google.com/document/d/1ngnuyt1f-MZMR2XC8z82kdpI_ERLKFQ_aotRAXTfPr8/edit)

#### Summary of changes 

Add name labels to the Circle of Trust view controller

#### Implementation Summary

Changes in `CircleOfTrustEditViewController`: 

- make `loadNumbers` a static function since it's used exact same way in `CircleOfTrustViewController`

Changes in `CircleOfTrustViewController`: 

- Add outlets for the name labels
- Add `nameLabels` array variable 
- Delete `loadNumbers` function and use static `CircleOfTrustEditViewController.loadNumbers()` instead 
- Rename `updateImageViews` and `loadContactPhotos` functions to `updateViews` and `loadContactPhotosAndNames`to reflect the changes of adding name labels to the view 
- Add `var nameImageDictionary = [Int: (String?, Data?)]()` dictionary variable where key is the index of a contact in the `names` array and value is a tuple for name sting and image data optional values. Key is later being used it as an index in the `imageViews` and `nameLabels` arrays to display corresponding name and image in the view. 

Changes in `Main.storyboard`: 

- add labels and layout constraints to correspond with each trustee image view 
- create outlet connections 
- increase image size for trustee image views and spacing between the views 
- rename screen title to "Circle of Trust" (instead of "Circle of Trusts")


#### How to test

Tests summary: Tested UI changes and edit numbers/display numbers flow, including contacts with images, in simulator and on iPhone 5 device.  

- Open the app 
- Login 
- Tap Circle of Trust 
- Tap Edit button 
- Select contact(s) from your phone's address book
- Tap Save 
- Check that the image views and name labels were updated accordingly 

![img_3185](https://user-images.githubusercontent.com/12476189/31982507-f64314a2-b90d-11e7-9886-fd75f536f2ed.PNG)
